app/models/annotation/AnnotationService.scala

```diff
@@ -246,7 +248,7 @@ object AnnotationService
     def getDatasetScale(dataSetName: String) = {
       for {
         dataSet <- DataSetDAO.findOneBySourceName(dataSetName)
-        scale <- dataSet.dataSource.scaleOpt
+        scale <- dataSet.dataSource.scaleOpt ?~> Messages("nml.scaleNotFound")
```

Comment:
Why do we have to duplicate this? Could we just have `scale` or `getScale` that resolves the option?

Reply:
Most of our accessors are designed to return options.
The `?~>` in a Fox-style for comprehension casts the option to a Fox and then uses the Fox method `?~>` to chain failure messages.

I’m not saying that this is the best design, but before we start changing that, we should discuss in more depth. So I’d suggest keeping it here for now, even if it brings some duplication.

Reply:
Absolutely agree.